homeassistant/components/zwave_js/cover.py

```diff
@@ -99,3 +111,57 @@ async def async_stop_cover(self, **kwargs: Any) -> None:
         target_value = self.get_zwave_value("Close") or self.get_zwave_value("Down")
         if target_value:
             await self.info.node.async_set_value(target_value, False)
+
+
+class ZwaveMotorizedBarrier(ZWaveBaseEntity, CoverEntity):
+    """Representation of a Z-Wave motorized barrier device."""
+
+    @property
+    def supported_features(self) -> Optional[int]:
+        """Flag supported features."""
+        return SUPPORT_OPEN | SUPPORT_CLOSE
+
+    @property
+    def device_class(self) -> Optional[str]:
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        return DEVICE_CLASS_GARAGE
+
+    @property
+    def is_opening(self) -> Optional[bool]:
+        """Return if the cover is opening or not."""
+        if self.info.primary_value.value is None:
+            return None
+        return bool(self.info.primary_value.value == BARRIER_STATE_OPENING)
+
+    @property
+    def is_closing(self) -> Optional[bool]:
+        """Return if the cover is closing or not."""
+        if self.info.primary_value.value is None:
+            return None
+        return bool(self.info.primary_value.value == BARRIER_STATE_CLOSING)
+
+    @property
+    def is_closed(self) -> Optional[bool]:
+        """Return if the cover is closed or not."""
+        if self.info.primary_value.value is None:
+            return None
+        # If a barrier is in the stopped state, the only way to proceed is by
+        # issuing an open cover command. Return None in this case which
+        # produces an unknown state and allows it to be resolved with an open
+        # command.
+        if self.info.primary_value.value == BARRIER_STATE_STOPPED:
```

Comment:
It would be good to add a test for this case as we expect this to happen whenever the cover is stopped.